packages/ag-charts-enterprise/src/features/color-picker/colorPicker.ts

```diff
@@ -22,6 +22,8 @@ const getHsva = (input: string) => {
 
 export class ColorPicker extends _ModuleSupport.BaseModuleInstance implements _ModuleSupport.ModuleInstance {
     private readonly element: HTMLElement;
+    private anchor?: { x: number; y: number };
```

Comment:
Maybe the anchor should be a rect, and we position ourself around that rect?

Reply:
The reasoning for using the fallback with optional `x` and `y` is because in this instance, we do not want it to jump to the left of the buttons. The behaviour differs between x & y.

If you provide an x and not a y, it slides along the y axis, but jumps on the x. In the case where we have a vertical list of buttons, we will want this behaviour.

In the annotations case it slides along the x axis, but jumps on the y.

Sliding x:
<img width="261" alt="Screenshot 2024-06-25 at 12 33 25" src="https://github.com/ag-grid/ag-charts/assets/3817697/f0cc6d49-89ff-4261-afd0-58fa9e64455d">

Jumping y:
<img width="303" alt="Screenshot 2024-06-25 at 12 33 32" src="https://github.com/ag-grid/ag-charts/assets/3817697/5fa8a34c-6298-4722-8fab-9f0a65fa2b9d">